### PR TITLE
Deprecate oc adm migrate

### PR DESCRIPTION
**Pull request by soltysh**
/assign @mfojtik

### CODE REVIEW

File: pkg/cli/admin/migrate/templateinstances/templateinstances.go [hunk 1]
```diff
@@ -86,10 +86,11 @@ func NewMigrateTemplateInstancesOptions(streams genericclioptions.IOStreams) *Mi
...
 	cmd := &cobra.Command{
-		Use:     name,
-		Short:   "Update TemplateInstances to point to the latest group-version-kinds",
-		Long:    internalMigrateTemplateInstancesLong,
-		Example: fmt.Sprintf(internalMigrateTemplateInstancesExample, fullName),
+		Use:        name,
+		Short:      "Update TemplateInstances to point to the latest group-version-kinds",
+		Long:       internalMigrateTemplateInstancesLong,
+		Example:    fmt.Sprintf(internalMigrateTemplateInstancesExample, fullName),
+		Deprecated: "it is not usable in self-managed 4.x cluster",
```
Comment (smarterclayton) [inline pkg/cli/admin/migrate/templateinstances/templateinstances.go]: This isn't actually true?

File: pkg/cli/admin/migrate/images/imagerefs.go [hunk 2]
```diff
@@ -89,10 +89,11 @@ func NewMigrateImageReferenceOptions(streams genericclioptions.IOStreams) *Migra
...
 	cmd := &cobra.Command{
-		Use:     fmt.Sprintf("%s REGISTRY/NAME=REGISTRY/NAME [...]", name),
-		Short:   "Update embedded Docker image references",
-		Long:    internalMigrateImagesLong,
-		Example: fmt.Sprintf(internalMigrateImagesExample, fullName),
+		Use:        fmt.Sprintf("%s REGISTRY/NAME=REGISTRY/NAME [...]", name),
+		Short:      "Update embedded Docker image references",
+		Long:       internalMigrateImagesLong,
+		Example:    fmt.Sprintf(internalMigrateImagesExample, fullName),
+		Deprecated: "it is not usable in self-managed 4.x cluster",
```
Comment (smarterclayton) [inline pkg/cli/admin/migrate/images/imagerefs.go]: This message doesn't make a ton of sense - might be more concise to say "migration of content is managed automatically in OpenShift 4" even if it's not quite true yet, some of these are.

Comment (sallyom) [reply pkg/cli/admin/migrate/images/imagerefs.go]: I'll open a follow-up to clarify

**Pull request merged**